packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -101,23 +101,23 @@ const String _kFlutterPluginsNameKey = 'name';
 const String _kFlutterPluginsPathKey = 'path';
 const String _kFlutterPluginsDependenciesKey = 'dependencies';
 
-  /// Filters [plugins] to those supported by [platformKey].
-  List<Map<String, dynamic>> _filterPluginsByPlatform(List<Plugin>plugins, String platformKey) {
-    final Iterable<Plugin> platformPlugins = plugins.where((Plugin p) {
-      return p.platforms.containsKey(platformKey);
+/// Filters [plugins] to those supported by [platformKey].
+List<Map<String, dynamic>> _filterPluginsByPlatform(List<Plugin> plugins, String platformKey) {
+  final Iterable<Plugin> platformPlugins = plugins.where((Plugin p) {
+    return p.platforms.containsKey(platformKey);
+  });
+
+  final Set<String> pluginNames = platformPlugins.map((Plugin plugin) => plugin.name).toSet();
+  final List<Map<String, dynamic>> list = <Map<String, dynamic>>[];
```

Comment:
nit: maybe rename something more descriptive like `plugins` while you're in here. Not a huge improvement, but a little better.

Reply:
The bigger issue is that this method is called "filter" but what it actually does is filter and tranform. But I'll call this `pluginInfo` to at least help highlight that the list isn't actually of `Plugin`s.

Reply:
pluginInfo is good; originally I'd though `platformPlugins`, but that's (a) taken and (b) only really a serialised representation of which platform plugins we're talking about :/